template.tpl

```diff
@@ -257,13 +257,47 @@ ___TEMPLATE_PARAMETERS___
     "groupStyle": "ZIPPY_CLOSED",
     "subParams": [
       {
-        "type": "CHECKBOX",
-        "name": "includeSelfDescribingEvent",
-        "checkboxText": "Include Self Describing event",
-        "simpleValueType": true,
-        "defaultValue": true,
-        "help": "Indicates if a Snowplow Self Describing event should be in the event object.",
-        "alwaysInSummary": true
+        "type": "GROUP",
+        "name": "selfDescribingEvent",
+        "displayName": "Snowplow Self Describing Event",
+        "groupStyle": "ZIPPY_OPEN",
+        "subParams": [
+          {
+            "type": "CHECKBOX",
+            "name": "includeSelfDescribingEvent",
+            "checkboxText": "Include Self Describing event",
+            "simpleValueType": true,
+            "alwaysInSummary": true,
+            "defaultValue": true,
+            "help": "Indicates if a Snowplow Self Describing event should be in the event object."
+          },
+          {
+            "type": "SELECT",
+            "name": "selfDescribingEventLocation",
+            "displayName": "Self Describing Event Location",
+            "macrosInSelect": false,
+            "selectItems": [
+              {
+                "value": "object",
+                "displayValue": "Schema name object on Braze Event"
+              },
+              {
+                "value": "root",
+                "displayValue": "SDE Event Properties added to Braze Event Root"
```

Comment:
These are pretty wordy for a select box. Could it just be like "Nested Object"  or "Merge to Root"?

Reply:
Changed them to:

- "Nest under schema name"
- "Merge to root level"

Maybe in the future we'd like to add an option to "Nest under custom path", that is why i specified the "under schema name" for the default option.

Also tweaked the help text to:
"Indicate the location where Snowplow Self Describing event properties should be added under Braze event properties.".

Do these sound ok? What do you think?

Reply:
Sounds good, thanks!